test/jasmine/tests/sankey_test.js

```diff
@@ -0,0 +1,226 @@
+
+                var fullTrace = _supply({});
+
+                expect(fullTrace.node.label)
+                    .toEqual([], 'presence of node label array is guaranteed');
+
+                expect(fullTrace.link.value)
+                    .toEqual([], 'presence of link value array is guaranteed');
+
+                expect(fullTrace.link.source)
+                    .toEqual([], 'presence of link source array is guaranteed');
+
+                expect(fullTrace.link.target)
+                    .toEqual([], 'presence of link target array is guaranteed');
+
+            });
+
+        it('\'Sankey\' specification should have proper types',
+            function() {
+
+                var fullTrace = _supply({});
+
+                expect(fullTrace.orientation)
+                    .toEqual(attributes.orientation.dflt, 'use orientation by default');
+
+                expect(fullTrace.valueformat)
+                    .toEqual(attributes.valueformat.dflt, 'valueformat by default');
+
+                expect(fullTrace.valuesuffix)
+                    .toEqual(attributes.valuesuffix.dflt, 'valuesuffix by default');
+
+                expect(fullTrace.arrangement)
+                    .toEqual(attributes.arrangement.dflt, 'arrangement by default');
+
+                expect(fullTrace.domain.x)
+                    .toEqual(attributes.domain.x.dflt, 'x domain by default');
+
+                expect(fullTrace.domain.y)
+                    .toEqual(attributes.domain.y.dflt, 'y domain by default');
+            });
+
+        it('\'Sankey\' layout dependent specification should have proper types',
+            function() {
+
+                var fullTrace = _supplyWithLayout({}, {font: {family: 'Arial'}});
+                expect(fullTrace.textfont)
+                    .toEqual({family: 'Arial'}, 'textfont is defined');
+            });
+
+        it('\'line\' specifications should yield the default values',
+            function() {
+
+                var fullTrace = _supply({});
+
+                expect(fullTrace.node.line.color)
+                    .toEqual('#444', 'default node line color');
+                expect(fullTrace.node.line.width)
+                    .toEqual(0.5, 'default node line thickness');
+
+                expect(fullTrace.link.line.color)
+                    .toEqual('#444', 'default link line color');
+                expect(fullTrace.link.line.width)
+                    .toEqual(0, 'default link line thickness');
+            });
+
+        it('fills \'node\' colors if not specified', function() {
+
+            var fullTrace = _supply({
+                node: {
+                    label: ['a', 'b']
+                },
+                link: {
+                    source: [0],
+                    target: [1],
+                    value: [1]
+                }
+            });
+
+            expect(Lib.isArray(fullTrace.node.color)).toBe(true, 'set up color array');
+
+        });
+
+    });
```

Comment:
We should at the very least add some `restyle` tests checking that we can toggle `visible: true || false` correctly and then sankey subplot get clear correctly on `deleteTraces`. Some hover labels test would be much appreciated too.